apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/TypeTests.cs

```diff
@@ -494,6 +494,87 @@ private void AssertJsonEqual<T>(string expected, string actual) where T: IMessag
             });
         }
 
+        // Note: Value.Parse fails for list values containing null, hence the separate tests
+        [Fact]
+        public void SerializeStringArrayContainingNull()
+        {
+            var input = new[] { "x", null, "y" };
+            var expected = new Value { ListValue = new ListValue { Values = { Value.ForString("x"), Value.ForNull(), Value.ForString("y") } } };
+            var options = SpannerConversionOptions.Default;
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.String).ToProtobufValue(input, options));
+        }
+
+        [Fact]
+        public void DeserializeStringArrayContainingNull()
+        {
+            var protobuf = new Value { ListValue = new ListValue { Values = { Value.ForString("x"), Value.ForNull(), Value.ForString("y") } } };
+            var options = SpannerConversionOptions.Default;
+            var expected = new[] { "x", null, "y" };
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.String).ConvertToClrType<string[]>(protobuf, options));
+        }
+
+        [Fact]
+        public void SerializeNullableDoubleArrayContainingNull()
+        {
+            // The null value is converted into a 0
+            var input = new double?[] { 5.5, null, 10.5 };
+            var expected = new Value { ListValue = new ListValue { Values = { Value.ForNumber(5.5), Value.ForNull(), Value.ForNumber(10.5) } } };
+            var options = SpannerConversionOptions.Default;
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.Float64).ToProtobufValue(input, options));
+        }
+
+        [Fact]
+        public void DeserializeNullableDoubleArrayContainingNull()
+        {
+            var protobuf = new Value { ListValue = new ListValue { Values = { Value.ForNumber(5.5), Value.ForNull(), Value.ForNumber(10.5) } } };
+            var options = SpannerConversionOptions.Default;
+            var expected = new double?[] { 5.5, null, 10.5 };
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.Float64).ConvertToClrType<double?[]>(protobuf, options));
+        }
+
+        [Fact]
+        public void DeserializeDoubleArrayContainingNull()
+        {
+            var protobuf = new Value { ListValue = new ListValue { Values = { Value.ForNumber(5.5), Value.ForNull(), Value.ForNumber(10.5) } } };
+            var options = SpannerConversionOptions.Default;
+
+            // The null value is converted into 0
+            var expected = new[] { 5.5, 0.0, 10.5 };
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.Float64).ConvertToClrType<double[]>(protobuf, options));
+        }
+
+        public static IEnumerable<object[]> UseClrDefaultForNulls { get; } = new[]
```

Comment:
What about `TheoryData<object, SpannerDbType, System.Type>`? It makes this line a bit longer, but the other below get a bit cleaner.

Reply:
Done.